test/irmin-pack/test_gc.ml

```diff
@@ -493,9 +506,195 @@ module Blocking_gc = struct
       tc "Test ro close" ro_close;
       tc "Test ro reload after open" ro_reload_after_v;
       tc "Test lru" gc_lru;
+      tc "Test gc during batch" gc_during_batch;
     ]
 end
 
 module Concurrent_gc = struct
-  let tests = []
+  (** Check that finding old objects during a gc works. *)
+  let find_during_gc ~lru_size () =
+    let* t = init ~lru_size () in
+    let* t, c1 = commit_1 t in
+    let* t = checkout_exn t c1 in
+    let* t, c2 = commit_2 t in
+    [%log.debug "Gc c1 keep c2"];
+    let* () = start_gc t c2 in
+    let* () = check_1 t c1 in
+    let* () = check_2 t c2 in
+    let* () = finalise_gc t in
+    let* () = check_not_found t c1 "removed c1" in
+    let* () = check_2 t c2 in
+    S.Repo.close t.repo
+
+  (** Check adding new objects during a gc and finding them after the gc. *)
+  let add_during_gc ~lru_size () =
+    let* t = init ~lru_size () in
+    let* t, c1 = commit_1 t in
+    let* t = checkout_exn t c1 in
+    let* t, c2 = commit_2 t in
+    [%log.debug "Gc c1 keep c2"];
+    let* () = start_gc t c2 in
+    let* t = checkout_exn t c2 in
+    let* t, c3 = commit_3 t in
+    let* () = finalise_gc t in
+    let* () = check_not_found t c1 "removed c1" in
+    let* () = check_2 t c2 in
+    let* () = check_3 t c3 in
```

Comment:
this fails, as the non blocking dummy gc does not have a way to handle the "newies" (objects added during gc).